src/Core/System/Capabilities.js

```diff
@@ -17,16 +34,39 @@ export default {
         const gl = renderer.context;
         maxTexturesUnits = gl.getParameter(gl.MAX_TEXTURE_IMAGE_UNITS);
 
-        const debugInfo = gl.getExtension('WEBGL_debug_renderer_info');
-        if (debugInfo !== null) {
-            const vendor = gl.getParameter(debugInfo.UNMASKED_VENDOR_WEBGL);
-            if (vendor.indexOf('mesa') > -1 || vendor.indexOf('Mesa') > -1) {
+        const program = gl.createProgram();
+        const glVertexShader = _WebGLShader(renderer, gl.VERTEX_SHADER, SampleTestVS);
+
+        let fragmentShader = `#define SAMPLE ${maxTexturesUnits}\n`;
+        fragmentShader += SampleTestFS;
+
+        const glFragmentShader = _WebGLShader(renderer, gl.FRAGMENT_SHADER, fragmentShader);
+
+        gl.attachShader(program, glVertexShader);
+        gl.attachShader(program, glFragmentShader);
+        gl.linkProgram(program);
+
+        if (gl.getProgramParameter(program, gl.LINK_STATUS) === false) {
+            if (maxTexturesUnits > 16) {
+                const info = gl.getProgramInfoLog(program);
+                // eslint-disable-next-line no-console
+                console.warn(`${info}: using a maximum of 16 texture units instead of the reported value (${maxTexturesUnits})`);
+                if (isFirefox()) {
+                    // eslint-disable-next-line no-console
+                    console.warn(`it can come from a bug mesa/firefox.
```

Comment:
Don't forget capitals ! Also I would write a slightly more understandable message:
> It can come from a Mesa/Firefox bug; the shader compiles to an error when using more than 16 sampler uniforms, see https://bugzilla.mozilla.org/show_bug.cgi?id=777028